### ISSUE TITLE
Broken runs when tag value contains new line characters

### ISSUE THREAD

**Issue by GCBallesteros** [mlflow/mlflow#479]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**:
No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Linux Ubuntu
- **MLflow installed from (source or binary)**: 
pip
- **MLflow version (run ``mlflow --version``)**:
0.6.0
- **Python version**: 
3.5.2

### Describe the problem
When a tag value contains a string with a new line character a generic exception is raised. If the run is done within a `with` statement with `start_run` after the exception is raised the run is left unfinished. As a result the the web UI is left in a broken state for the whole experiment in which the run was made.

I would expect tags to be able to contain arbitrary strings and broken runs to not completely break the web UI or at least to give a brief explanation as to what went wrong so that the faulty run can be manually removed from the `mlruns` directory.

### Source code / logs
```
with flow.start_run(experiment_id=3):
    extra_comments="""New lines break the tag, the run and the UI.
Oopsie."""
    mlflow.set_tag('ExtraComments', extra_comments)
```

```
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
<ipython-input-4-ddad37fe21b6> in <module>()
      3     extra_comments="""New lines break the tag, the run and UI.
      4 Oopsie."""
----> 5     mlflow.set_tag('ExtraComments', extra_comments)

/usr/local/lib/python3.5/dist-packages/mlflow/tracking/fluent.py in __exit__(self, exc_type, exc_val, exc_tb)
     38     def __exit__(self, exc_type, exc_val, exc_tb):
     39         status = "FINISHED" if exc_type is None else "FAILED"
---> 40         end_run(status)
     41         return exc_type is None
     42 

/usr/local/lib/python3.5/dist-packages/mlflow/tracking/fluent.py in end_run(status)
    116     global _active_run
    117     if _active_run:
--> 118         MlflowClient().set_terminated(_active_run.info.run_uuid, status)
    119         # Clear out the global existing run environment variable as well.
    120         env.unset_variable(_RUN_ID_ENV_VAR)

/usr/local/lib/python3.5/dist-packages/mlflow/tracking/client.py in set_terminated(self, run_id, status, end_time)
    202         status = status if status else "FINISHED"
    203         self.store.update_run_info(run_id, run_status=RunStatus.from_string(status),
--> 204                                    end_time=end_time)
    205 
    206 

/usr/local/lib/python3.5/dist-packages/mlflow/store/file_store.py in update_run_info(self, run_uuid, run_status, end_time)
    195     def update_run_info(self, run_uuid, run_status, end_time):
    196         _validate_run_id(run_uuid)
--> 197         run_info = self.get_run(run_uuid).info
    198         new_info = run_info._copy_with_overrides(run_status, end_time)
    199         run_dir = self._get_run_dir(run_info.experiment_id, run_info.run_uuid)

/usr/local/lib/python3.5/dist-packages/mlflow/store/file_store.py in get_run(self, run_uuid)
    247         metrics = self.get_all_metrics(run_uuid)
    248         params = self.get_all_params(run_uuid)
--> 249         tags = self.get_all_tags(run_uuid)
    250         return Run(run_info, RunData(metrics, params, tags))
    251 

/usr/local/lib/python3.5/dist-packages/mlflow/store/file_store.py in get_all_tags(self, run_uuid)
    358         tags = []
    359         for tag_file in tag_files:
--> 360             tags.append(self._get_tag_from_file(parent_path, tag_file))
    361         return tags
    362 

/usr/local/lib/python3.5/dist-packages/mlflow/store/file_store.py in _get_tag_from_file(parent_path, tag_name)
    336         if len(tag_data) > 1:
    337             raise Exception("Unexpected data for tag '%s'. Tag recorded more than once"
--> 338                             % tag_name)
    339         return RunTag(tag_name, str(tag_data[0].strip()))
    340 

Exception: Unexpected data for tag 'ExtraComments'. Tag recorded more than once
```

Thanks for sharing this great work with all of us!

**Comment by aadamson** [comment]
Hi @GCBallesteros, this issue should be fixed in master. [This patch](https://github.com/mlflow/mlflow/commit/c0fa23ef7017c459d4f77231238f910db0c4fbf0#diff-73334c51ec814d7e8b40d09d7d828f82L331) changes the mutliline tag behavior:

**Comment by GCBallesteros** [comment]
Thanks for the quick reply. I'll wait for the next release and in the mean time fix things manually.

**Comment by aarondav** [comment]
Closing since it's fixed in c0fa23ef7017c459d4f77231238f910db0c4fbf0 -- reopen if the problem is not fixed.
